File: backoffice/src/api/landing.ts
import client from "./client";

export type LandingContent = {
  id: number;
  name: string;
  secao: string;
  titulo: string;
  subtitulo: string;
  descricao: string;
  imagem: string;
  botaoTexto: string;
  botaoLink: string;
  status: string;
};

export type LandingContentInput = Omit<LandingContent, "id">;

export const landingAPI = {
  getAll: async (): Promise<{ contents: LandingContent[] }> => {
    const res = await client.get<{ contents: LandingContent[] }>("/admin/landing");
    return res.data;
  },

  create: async (input: LandingContentInput): Promise<{ content: LandingContent }> => {
    const res = await client.post<{ content: LandingContent }>("/admin/landing", input);
    return res.data;
  },

  update: async (
    id: number,
    input: LandingContentInput,
  ): Promise<{ content: LandingContent }> => {
    const res = await client.put<{ content: LandingContent }>(`/admin/landing/${id}`, input);
    return res.data;
  },

  delete: async (id: number): Promise<{ message: string }> => {
    const res = await client.delete<{ message: string }>(`/admin/landing/${id}`);
    return res.data;
  },
};

File: backend/internal/models/landing.go
package models

type LandingContent struct {
	ID         uint   `gorm:"primaryKey;autoIncrement" json:"id"`
	Name       string `gorm:"not null"                 json:"name"`
	Secao      string `gorm:"not null"                 json:"secao"`
	Titulo     string `gorm:"not null"                 json:"titulo"`
	Subtitulo  string `                                 json:"subtitulo"`
	Descricao  string `gorm:"type:text"                json:"descricao"`
	Imagem     string `gorm:"type:text"                json:"imagem"`
	BotaoTexto string `                                json:"botaoTexto"`
	BotaoLink  string `                                json:"botaoLink"`
	Status     string `gorm:"not null;default:'Ativo'" json:"status"`
}

func (LandingContent) TableName() string { return "landing_contents" }

type LandingContentInput struct {
	Name       string `json:"name"       binding:"required"`
	Secao      string `json:"secao"      binding:"required"`
	Titulo     string `json:"titulo"     binding:"required"`
	Subtitulo  string `json:"subtitulo"`
	Descricao  string `json:"descricao"`
	Imagem     string `json:"imagem"`
	BotaoTexto string `json:"botaoTexto"`
	BotaoLink  string `json:"botaoLink"`
	Status     string `json:"status"`
}
